core/shared/src/main/scala/scalaz/zio/duration/DurationConversions.scala

```diff
@@ -0,0 +1,37 @@
+package scalaz.zio.duration
+
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeUnit._
+
+trait DurationConversions extends Any {
+
+  protected def asDuration(unit: TimeUnit): Duration
```

Comment:
Instead of using inheritance here, let's make this a class:

```scala
class DurationSyntax(n: Long) {
  def nanoseconds = ...
```

Then the implicits will create wrappers of this class type.

We should rename the file to `DurationSyntax.scala`.